lib/compat/experimental/blocks.php

```diff
@@ -60,7 +60,12 @@ function build_comment_query_vars_from_block( $block ) {
 			} elseif ( 'oldest' === $default_page ) {
 				$comment_args['paged'] = 1;
 			} elseif ( 'newest' === $default_page ) {
-				$comment_args['paged'] = ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+				$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
```

Comment:
Type casting the `max_num_page` property is appropriate here. Let me explain.

The `max_num_pages` property is declared as an `integer`. BUT it's calculation uses `ceil` https://www.php.net/manual/en/function.ceil.php which is then `float`. This is an issue in WordPress Core's `WP_Comment_Query` class. But as it's a query class, likely it's not "quickly" fixable. Why? Any changes to anything query move slowly on purpose to maximize soak time (for testing and feedback). Why? Query functionality is mission critical and any changes are risky at the scale of 43% of the web. So by design changes happen slowly (often over several release cycles) in WP Core.

Is float to int conversion an issue in PHP 8.1+? No, not when it's an explicit `int` casting. No deprecation notice will be thrown and currently there are no plans for fatal error when PHP 9 lands in the future. See details here https://php.watch/versions/8.1/deprecate-implicit-conversion-incompatible-float-string.